gnomad/utils/filtering.py

```diff
@@ -531,3 +531,57 @@ def split_vds_by_strata(
     return {
         strata: hl.vds.filter_samples(vds, list(s)) for strata, s in s_by_strata.items()
     }
+
+
+def filter_freq_by_meta(
+    freq_expr: hl.expr.ArrayExpression,
+    freq_meta_expr: hl.expr.ArrayExpression,
+    items_to_filter: Union[Dict[str, List[Any]], List[Any]],
+    keep: bool = True,
+    operator: str = "and",
+) -> [hl.expr.ArrayExpression, hl.expr.ArrayExpression]:
+    """
+    Filter frequency and frequency meta expressions by freq_meta items.
+
+    This function is designed to filter in different cases, for example:
+    simply filter by a list of keys, e.g. ["sex", "downsampling"],
+    or by specific populations by using {"pop": ["han", "papuan"]},
+    or a more complicated use case: {"pop": ["afr"], "sex": ["XX"]},
+    one can decide to keep or remove the items, either they appear
+    at the same time in one freq_meta dictionary by using "and",
+    or they appear in different freq_meta dictionaries by using "or".
+
+    :param freq_expr: frequency expression
+    :param freq_meta_expr: frequency meta expression
+    :param items_to_filter: items to filter by, either a list or a dictionary
+    :param keep: whether to keep or remove the items
+    :param operator: whether to use "and" or "or" to combine the items
+    :return: filtered frequency and frequency meta expressions
+    """
+    freq_meta_expr = freq_meta_expr.collect(_localize=False)[0]
+
+    if operator == "and":
+        operator_func = hl.all
+    elif operator == "or":
+        operator_func = hl.any
+
+    if isinstance(items_to_filter, list):
+        filter_func = lambda m, k: m.contains(k)
+    elif isinstance(items_to_filter, dict):
+        filter_func = lambda m, k: (m.get(k[0], "") == k[1])
+        items_to_filter = [
+            (k, v) for k, values in items_to_filter.items() for v in values
+        ]
```

Comment:
I realized that I did this wrong and it won't work for certain cases, here and below are the fixes:
```suggestion
    if isinstance(items_to_filter, list):
        filter_func = lambda m, k: m.contains(k)
        items_to_filter = [[k] for k in items_to_filter]
    elif isinstance(items_to_filter, dict):
        filter_func = lambda m, k: (m.get(k[0], "") == k[1])
        items_to_filter = [
            [(k, v) for v in values] for k, values in items_to_filter.items()
        ]
```